src/rgw/rgw_admin.cc

```diff
@@ -11187,38 +11187,45 @@ int main(int argc, const char **argv)
     }
 
     formatter->open_object_section("result");
-    formatter->open_array_section("topics");
+    rgw_pubsub_topics result;
+    auto is_v2_ready = rgw::all_zonegroups_support(*site, rgw::zone_features::notification_v2) &&
+                       driver->stat_topics_v1(tenant, null_yield, dpp()) == -ENOENT;
+    if (is_v2_ready) {
+      formatter->open_array_section("topics");
+    }
     do {
-      rgw_pubsub_topics result;
       int ret = ps.get_topics(dpp(), next_token, max_entries,
                               result, next_token, null_yield);
       if (ret < 0 && ret != -ENOENT) {
         cerr << "ERROR: could not get topics: " << cpp_strerror(-ret) << std::endl;
         return -ret;
       }
+      if (!is_v2_ready) {
```

Comment:
does this mean thta v1 topics don't support paging?

Reply:
yes it doesn't support paging:
```
  if (!use_notification_v2 || driver->stat_topics_v1(tenant, y, dpp) != -ENOENT) {
    // in case of v1 or during migration we use v1 topics
    // v1 returns all topics, ignoring marker/max_items
    return read_topics_v1(dpp, result, nullptr, y);
  }
```

Reply:
in this case it would be simpler to separate the logic.
code would be simpler, and it would be clear that there is no paging in v1 topics